### PR TITLE
CHKO: Change remoteEnabled flag to checkType enum

### PR DESCRIPTION
**Pull request by burcuka**

### CODE REVIEW

**Review by schneidermic0 — APPROVED** [review]
Looks good to me

**Comment by albertmink** [comment]
And removes 'technical_id' :)

File: file-formats/chko/type/zif_aff_chko_v1.intf.abap [hunk 1]
```diff
@@ -43,11 +44,23 @@ INTERFACE zif_aff_chko_v1
...
+
+  CONSTANTS:
+    "! <p class="shorttext">Check Type</p>
+    "! Check Type
+    BEGIN OF co_check_type,
+      "! <p class="shorttext">Local</p>
+      "! Local
+      local              TYPE ty_check_type VALUE 'L',
+      "! <p class="shorttext">Remote Enabled</p>
+      "! Remote Enabled
```
Comment (albertmink) [inline file-formats/chko/type/zif_aff_chko_v1.intf.abap:63]:
```suggestion
      "! Remote enabled
```

File: file-formats/chko/type/zif_aff_chko_v1.intf.abap [hunk 2]
```diff
@@ -43,11 +44,23 @@ INTERFACE zif_aff_chko_v1
...
+      "! The check type of ATC check
+      check_type     TYPE ty_check_type,
       "! <p class="shorttext">Parameters</p>
       "! Parameters of the ATC check object
       parameters         TYPE ty_parameters,
     END OF ty_main.
+
+  CONSTANTS:
+    "! <p class="shorttext">Check Type</p>
+    "! Check Type
```
Comment (albertmink) [inline file-formats/chko/type/zif_aff_chko_v1.intf.abap:57]:
```suggestion
    "! Check type
```

File: file-formats/chko/type/zif_aff_chko_v1.intf.abap [hunk 3]
```diff
@@ -5,25 +5,26 @@ INTERFACE zif_aff_chko_v1
...
 
   TYPES:
     "! <p class="shorttext">Parameters</p>
     "! List of parameters
-    ty_parameters TYPE HASHED TABLE OF ty_parameter WITH UNIQUE KEY technical_id
-      WITH UNIQUE HASHED KEY xml_name COMPONENTS name.
+    ty_parameters TYPE HASHED TABLE OF ty_parameter WITH UNIQUE KEY name.
+
+  "! <p class="shorttext">Check Type</p>
+  "! Check Type
```
Comment (albertmink) [inline file-formats/chko/type/zif_aff_chko_v1.intf.abap:25]:
```suggestion
  "! Check type
```

**Review by albertmink — CHANGES_REQUESTED** [review]
The descriptions are non-titlecase.

File: file-formats/chko/type/zif_aff_chko_v1.intf.abap [hunk 4]
```diff
@@ -5,25 +5,26 @@ INTERFACE zif_aff_chko_v1
...
-      "! The technical Id of a parameter
-      technical_id TYPE c LENGTH 4,
       "! <p class="shorttext">Parameter Name</p>
       "! Name of a parameter
       name         TYPE string,
       "! <p class="shorttext">Parameter Description</p>
       "! Description of a parameter
       description  TYPE c LENGTH 255,
-      "! <p class="shorttext">Hidden Flag</p>
+      "! <p class="shorttext">Is hidden</p>
```
Comment (albertmink) [inline file-formats/chko/type/zif_aff_chko_v1.intf.abap:14]:
```suggestion
      "! <p class="shorttext">Is Hidden</p>
```

File: file-formats/chko/chko-v1.json [hunk 5]
```diff
@@ -102,7 +108,7 @@
             "maxLength": 255
           },
           "hidden": {
-            "title": "Hidden Flag",
+            "title": "Is hidden",
```
Comment (albertmink) [inline file-formats/chko/chko-v1.json:111]:
```suggestion
            "title": "Is Hidden",
```

**Review by albertmink — CHANGES_REQUESTED** [review]
Ahh, I found another case miss match 😇

**Review by albertmink — APPROVED** [review]

**Pull request merged**